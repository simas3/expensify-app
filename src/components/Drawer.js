import React from 'react';
import map from 'lodash/map'
import { List } from 'material-ui/List'
import Drawer from 'material-ui/Drawer'
import TopBar from './AppBar'
import links from '../data/linksData'
import SideBarItem from './SideBarItem'

export default class AppBarDrawer extends React.Component {

    constructor(props) {
        super(props);
        this.state = { open: false };
    }


    handleToggle = () => this.setState({ open: !this.state.open });

    handleClose = () => this.setState({ open: false });

    render() {
        return (
            <div>
                <TopBar handleToggle={this.handleToggle} />
                <Drawer
                    docked={false}
                    width={200}
                    open={this.state.open}
                    onRequestChange={(open) => this.setState({ open })}

                >
                    <List>
                        {map(links, ({ isExact, linkTo, text }, key) => {
                            return (
                                <SideBarItem
                                    isExact={isExact}
                                    linkTo={linkTo}
                                    primaryText={text}
                                    onClick={this.handleClose}
                                    key={key}
                                />
                            )
                        })}
                    </List>
                </Drawer>
            </div>
        );
    }
}