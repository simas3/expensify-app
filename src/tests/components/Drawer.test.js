import React from 'react'
import ReactShallowRenderer from 'react-test-renderer/shallow'
import Drawer from '../../components/Drawer'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json';

test('should render Drawer component', () => {
    const wrapper = shallow(<Drawer />)
    expect(toJson(wrapper)).toMatchSnapshot()
})