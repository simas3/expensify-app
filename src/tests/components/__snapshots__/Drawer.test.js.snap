// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Drawer component 1`] = `
<div>
  <TopBar
    handleToggle={[Function]}
  />
  <Drawer
    disableSwipeToOpen={false}
    docked={false}
    onRequestChange={[Function]}
    open={false}
    openSecondary={false}
    swipeAreaWidth={30}
    width={200}
    zDepth={2}
  >
    <List>
      <SideBarItem
        isExact={true}
        key="0"
        linkTo="/"
        onClick={[Function]}
        primaryText="Home"
      />
      <SideBarItem
        isExact={false}
        key="1"
        linkTo="/create"
        onClick={[Function]}
        primaryText="Create"
      />
      <SideBarItem
        isExact={false}
        key="2"
        linkTo="/help"
        onClick={[Function]}
        primaryText="Help"
      />
    </List>
  </Drawer>
</div>
`;
